flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/SessionContext.java

```diff
@@ -0,0 +1,303 @@
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.factories.PlannerFactoryUtil;
+import org.apache.flink.table.gateway.common.endpoint.EndpointVersion;
+import org.apache.flink.table.gateway.common.session.SessionHandle;
+import org.apache.flink.table.gateway.service.operation.OperationManager;
+import org.apache.flink.table.module.ModuleManager;
+import org.apache.flink.util.TemporaryClassLoaderContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Context describing a session, it's mainly used for user to open a new session in the backend. If
+ * client request to open a new session, the backend {@code Executor} will maintain the session
+ * context map util users close it.
+ */
+public class SessionContext {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SessionContext.class);
+
+    private final SessionHandle sessionId;
+    private final EndpointVersion endpointVersion;
+
+    // store all options and use Configuration to build SessionState and TableConfig.
+    private final Configuration sessionConf;
+    private final SessionState sessionState;
+    private final URLClassLoader userClassloader;
+
+    private final OperationManager operationManager;
+
+    private SessionContext(
+            SessionHandle sessionId,
+            EndpointVersion endpointVersion,
+            Configuration sessionConf,
+            URLClassLoader classLoader,
+            SessionState sessionState,
+            OperationManager operationManager) {
+        this.sessionId = sessionId;
+        this.endpointVersion = endpointVersion;
+        this.sessionConf = sessionConf;
+        this.userClassloader = classLoader;
+        this.sessionState = sessionState;
+        this.operationManager = operationManager;
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Getter method
+    // --------------------------------------------------------------------------------------------
+
+    public SessionHandle getSessionId() {
+        return this.sessionId;
+    }
+
+    public Map<String, String> getConfigMap() {
+        return sessionConf.toMap();
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Method to execute commands
+    // --------------------------------------------------------------------------------------------
+
+    /** Close resources, e.g. catalogs. */
+    public void close() {
+        try (TemporaryClassLoaderContext ignored =
+                TemporaryClassLoaderContext.of(userClassloader)) {
```

Comment:
I think we can't. Because CatalogManager doesn't hold the user classloader now and uses the default classloder to load the class when unregistering the catalog.